test/helpers/form_helper_test.rb

```diff
@@ -0,0 +1,61 @@
+require 'test_helper'
+
+class FormHelperTest < ActionView::TestCase
```

Comment:
I like this suite of tests - very thorough and satisfying to understand :-)

Reply:
Moving that markup to a helper made this much easier to test too!